MonstieWash/Assets/Scripts/MonsterAi/MonsterBrain.cs

```diff
@@ -293,6 +295,15 @@ private void CalculateAggression()
         m_lastAttackTime = 0f;
         m_attackTimer = UnityEngine.Random.Range(minBetweenAttacks, maxBetweenAttacks);
         MonsterAttack?.Invoke();
+
+        var monsterController = FindFirstObjectByType<MonsterController>();
+        Action onAttackComplete = null;
+        onAttackComplete = delegate ()
+        {
+            monsterController.OnAttackEnd -= onAttackComplete;
+            SpawnDirt();
```

Comment:
I might be wrong but I think this should instead be called from the player hand taking damage.
As currently stands even if the player dodges the attack the slime will receive new dirt patches.
Not sure of intended interaction though designers may want that.

Reply:
yeah it said "on attack" so I assumed it was for every attack since it'll be made that it only attacks when you make it flinch enough times. I'll double check with designers tho.